src/function.py

```diff
@@ -122,6 +123,12 @@ def zip_and_upload_folder(client: CogniteClient, config: FunctionConfig, name: s
 @retry(exceptions=(IOError, FunctionDeployTimeout, FunctionDeployError), tries=5, delay=2, jitter=2)
 def upload_and_create(client: CogniteClient, config: FunctionConfig) -> Function:
     zip_file_name = get_file_name(config.external_id)  # Also external ID
+
+    if config.overwrite:
+        # upsert was requested. delete schedules, function and files
+        try_delete_function(client=client, external_id=config.external_id)
+        try_delete_function_file(client=client, external_id=config.external_id)
```

Comment:
both calls do client.<>.retrieve which should raise if the function is missing.... AFAIK the code... but should we just in case to swallow exceptions here?

Reply:
Both uses "resource_exists" before attempting, so I think we should not catch any errors here